public/js/dom.js

```diff
@@ -0,0 +1,104 @@
+const resultContainer = document.querySelector(".result-container");
+
+const searchcontainer = document.querySelector(".search-result");
+const serchInput = document.getElementById("search");
+const reposDiv = document.createElement("div");
+
+reposDiv.classList.add("repos");
+resultContainer.appendChild(reposDiv);
+
+const delayKeyUp = (() => {
+  let timer = null;
+  const delay = (func, ms) => {
+    timer ? clearTimeout(timer) : null;
+    timer = setTimeout(func, ms);
+  };
+  return delay;
+})();
+
+serchInput.addEventListener("keyup", (e) => {
+  const query = e.target.value;
+  delayKeyUp(() => {
+    if (query === "") {
+      searchcontainer.textContent = "";
+    } else {
+      fetch(`/users/search/${query}`)
+        .then((data) => data.json())
+        .then((data) => {
+          rendersearchResult(data);
+        })
+        .catch((err) => {
+          console.log(err);
```

Comment:
what if the fetch fails then `rendersearchResult` will not receive the expected value for `data` and will cause some error, so you should handle such cases by displaying a nice message for the user ex `no result found` `try another search term`

**NOTE** this thing about how the fetch api deals with request/response errors
```js
fetch(url).then((response) => {
  // Always gets a response, unless there is network error
  // It never throws an error for 4xx or 5xx response 😟
 // so here we need to check for the response status and display the result if it is 2xx or display error message if it is 4xx or 5xx
}).catch((error) => {
  // Only network error comes here
});
```